simulator/src/main/java/com/hazelcast/simulator/worker/WorkerMessageProcessor.java

```diff
@@ -92,22 +91,18 @@ private boolean isMaster() {
                 public Boolean call() throws Exception {
                     return isOldestMember();
                 }
-            }, 10, TimeUnit.SECONDS).get(TIMEOUT, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-            throw new IllegalStateException(e);
-        } catch (ExecutionException e) {
-            throw new IllegalStateException(e);
-        } catch (TimeoutException e) {
+            }, delaySeconds, TimeUnit.SECONDS).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (Exception e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private boolean isOldestMember() {
+    boolean isOldestMember() {
```

Comment:
well, it breaks encapsulation. as you are effectively making the methods to be part of the API (even on just package level). 

I think in this case the best thing to do is to pull out the method to a util class. as that's why they are anyway - checking the oldest member shouldn't be a responsibility of the WMProcessor.